src/directory_schema/directory_schema.py

```diff
@@ -0,0 +1,111 @@
+import os
+
+from jsonschema import Draft7Validator, validate, ValidationError
+from yaml import dump as dump_yaml
+
+
+def to_dir_listing(dir_as_list, indent=''):
+    next_indent = indent + '    '
+    return ''.join([
+        '\n' + indent + item['name']
+        + to_dir_listing(
+            item['contents'] if 'contents' in item else [],
+            next_indent
+        )
+        for item in dir_as_list
+    ])
+
+
+def dir_to_dict(path):
+    '''
+    Walk the directory at `path`, and return a dict like that from `tree -J`:
```

Comment:
Doesn't this return a `list`? Could you maybe clarify the below example if that is the case to make it clear what a nested structure would look like?

Reply:
No, you're right; I'll fix. I keep getting the dict with the directory's name confused with the list that has its contents.